core/network-libp2p/src/custom_proto/behaviour.rs

```diff
@@ -313,9 +313,8 @@ impl<TMessage, TSubstream> CustomProto<TMessage, TSubstream> {
 	}
 
 	/// Indicates to the peerset that we have discovered new addresses for a given node.
-	pub fn add_discovered_node(&mut self, peer_id: &PeerId) {
-		debug!(target: "sub-libp2p", "PSM <= Discovered({:?})", peer_id);
```

Comment:
not sure what to do about this `debug!`. I could move print it in `Peerset::discovered` if needed

Reply:
I think it would be nice to have the logging both in `sub-libp2p` and in the peerset.
I guess we could have a custom iterator that does the same as `Iterator::map` but also implements `Drop` and prints the elements that are ignored?

Reply:
Thanks for suggestion. I realised that we can just print all items inside `.map` combinator. I fixed it in the latest commit